src/js/components/productSupplier/ProductSupplierListTable.jsx

```diff
@@ -1,27 +1,103 @@
+import React, { useMemo, useState } from 'react';
 
 import PropTypes from 'prop-types';
+import { RiDeleteBinLine, RiPencilLine } from 'react-icons/ri';
+import { useSelector } from 'react-redux';
 
+import productSupplierApi from 'api/services/ProductSupplierApi';
 import DataTable, { TableCell } from 'components/DataTable';
 import DateCell from 'components/DataTable/DateCell';
 import PreferenceTypeColumn from 'components/productSupplier/PreferenceTypeColumn';
+import { PRODUCT_SUPPLIER_URL } from 'consts/applicationUrls';
 import useProductSupplierListTableData from 'hooks/list-pages/productSupplier/useProductSupplierListTableData';
+import ActionDots from 'utils/ActionDots';
+import CustomModal from 'utils/CustomModal';
+import { hasPermissionsToProductSourceActions } from 'utils/permissionUtils';
 import StatusIndicator from 'utils/StatusIndicator';
 import Translate from 'utils/Translate';
 import ListTableTitleWrapper from 'wrappers/ListTableTitleWrapper';
 import ListTableWrapper from 'wrappers/ListTableWrapper';
 
 const ProductSupplierListTable = ({ filterParams }) => {
+  const [isDeleteConfirmationOpened, setIsDeleteConfirmationOpened] = useState(false);
+  const [selectedProductSupplierId, setSelectedProductSupplierId] = useState(null);
+
+  const { currentUser } = useSelector((state) => ({
+    currentUser: state.session.user,
+  }));
+
+  const closeDeleteConfirmationModal = () => {
+    setIsDeleteConfirmationOpened(false);
+  };
+
+  const deleteProductSupplier = async () => {
+    try {
+      await productSupplierApi.deleteProductSupplier(selectedProductSupplierId);
+    } finally {
+      setIsDeleteConfirmationOpened(false);
+    }
+  };
+
+  const deleteConfirmationModalButtons = [
+    {
+      variant: 'transparent',
+      defaultLabel: 'Cancel',
+      label: 'Cancel',
+      onClick: closeDeleteConfirmationModal,
+    },
+    {
+      variant: 'danger',
+      defaultLabel: 'Delete',
+      label: '',
+      onClick: deleteProductSupplier,
+    },
+  ];
+
+  const getActions = (id) => [
+    {
+      defaultLabel: 'Edit',
+      label: 'react.productSupplier.edit.label',
+      leftIcon: <RiPencilLine />,
+      onClick: () => {
+        window.location = PRODUCT_SUPPLIER_URL.edit(id);
+      },
+    },
+    {
+      defaultLabel: 'Delete Product Source',
+      label: 'react.productSupplier.delete.label',
+      leftIcon: <RiDeleteBinLine />,
+      variant: 'danger',
+      onClick: () => {
+        setIsDeleteConfirmationOpened(true);
+        setSelectedProductSupplierId(id);
+      },
+    },
+  ];
+
   const columns = useMemo(() => [
     {
       Header: ' ',
       width: 50,
-      fixed: 'left',
       sortable: false,
       style: {
         overflow: 'visible',
         zIndex: 1,
       },
+      fixed: 'left',
+      Cell: (row) => {
+        const clickableActions = hasPermissionsToProductSourceActions(currentUser)
+          ? { actions: getActions(row.original.id) }
+          : {};
```

Comment:
I don't mind this version, but couldn't we potentially refactor the `findActions` util to include the new condition (roles)?
There is also a possibility to pass `customFilter` to `findActions` so maybe it's just a matter of passing your `hasPermissionsToProductSourceActions` as `customFilter` prop.